tests/scripts/check_test_cases.py

```diff
@@ -111,6 +112,24 @@ def walk_ssl_opt_sh(self, file_name):
                 self.process_test_case(descriptions,
                                        file_name, line_number, description)
 
+    def walk_compat_sh(self, file_name):
+        """Iterate over the test cases compat.sh with a similar format."""
+        descriptions = self.new_per_file_state() # pylint: disable=assignment-from-none
+        compat_cmd = ['sh', file_name, '--list-test-case']
+        result = subprocess.run(compat_cmd,
+                                stdout=subprocess.PIPE,
+                                check=False)
+        if result.returncode != 0:
+            print(*compat_cmd, 'returned', str(result.returncode))
+            return
+        else:
+            # Pattern: g->m dtls12,no TLS_DHE_PSK_WITH_AES_128_CBC_SHA .......... \n
+            m = re.findall(br'[^ogm]*((?:[ogm]->[ogm]\s*\w*.\w*\s\w*)*)\s*\.*\s*\n',
+                           result.stdout)
```

Comment:
I had some discussion with @lpy4105 offline. I think current design is problematic as we're supposed to maintain test case description `$TITLE` in `compat.sh` instead of `check_test_case.py` via regex expression. Therefore we'd better call a function to uniform `$TITLE` used by `--list-test-case`, `run_client` and `$MBEDTLS_TEST_OUTCOME_FILE` (in #7136). As a result, we can simply split `result.stdout` into List to store test case description correspondingly. Let me know if there are better options.